circuit_maintenance_parser/provider.py

```diff
@@ -59,12 +69,44 @@ class GenericProvider(BaseModel):
     _processors: List[GenericProcessor] = [SimpleProcessor(data_parsers=[ICal])]
     _default_organizer: str = "unknown"
 
+    _include_filter: Dict[str, List[str]] = {}
+    _exclude_filter: Dict[str, List[str]] = {}
+
+    def include_filter_check(self, data: NotificationData) -> bool:
+        """If `_include_filter` is defined, it verifies that the matching criteria is met."""
+        if self._include_filter:
+            return self.filter_check(self._include_filter, data)
+        return True
+
+    def exclude_filter_check(self, data: NotificationData) -> bool:
+        """If `_exclude_filter` is defined, it verifies that the matching criteria is met."""
+        if self._exclude_filter:
+            return self.filter_check(self._exclude_filter, data)
+        return False
+
+    @staticmethod
+    def filter_check(filter_dict: Dict, data: NotificationData) -> bool:
+        """Generic filter check."""
+        for data_part in data.data_parts:
+            filter_data_type = data_part.type
+            if filter_data_type not in filter_dict:
+                continue
+
+            data_part_content = data_part.content.decode()
+            if any(re.search(filter_re, data_part_content) for filter_re in filter_dict[filter_data_type]):
+                return True
```

Comment:
Would probably be helpful to log a debug message when a match is found, to aid in troubleshooting as to why a particular message was or wasn't filtered.